roles/vcenter_host_connection/README.md

```diff
@@ -0,0 +1,124 @@
+# vcenter_host_connection
+
+Add, remove, connect, disconnect, or reconnect an ESXi host from a VCenter
+
+## Requirements
+
+pyvomi
+
+## Role Variables
+
+### Auth
+- **vcenter_host_connection_hostname**:
+  - str, The hostname of the vcenter to which you want to connect. Required
+
+- **vcenter_host_connection_username**:
+  - str, The username to use to authenticate to the vcenter to which you want to connect. Required
+
+- **vcenter_host_connection_password**:
+  - str, The password to use to authenticate to the esxi or vcenter to which you want to connect. Required
+
+- **vcenter_host_connection_validate_certs**:
+  - bool, If true then certificates will be validated when connecting to the vcenter for auth. Optional.
+
+- **vcenter_host_connection_port**:
+  - int, The port of the vcenter to which you want to connect. Optional.
+
+### Placement
+- **vcenter_host_connection_folder**:
+  - str, The folder path where the ESXi host should be added. Required if the cluster name is not provided
+
+- **vcenter_host_connection_datacenter**:
+  - str, The datacenter name where the ESXi host should be added. Required
+
+- **vcenter_host_connection_cluster**:
+  - str, The cluster name where the ESXi host should be added. Required if the folder name is not provided
+
+### Connection Settings
+- **vcenter_host_connection_state**:
+  - str, The connection state of the ESXi host that you want to set. Default is `present`
+  - If set to `present`, add the host if host is absent, or update the location of the host if host already exists.
+  - If set to `absent`, remove the host if host is present, or do nothing if host already does not exists.
+  - If set to `add_or_reconnect`, add the host if it's absent else reconnect it and update the location.
+  - If set to `reconnect`, then reconnect the host if it's present and update the location.
```

Comment:
from @machacekondra 

> Could this be just a parameter, instead of state. It would work only with present state.

I copied this section from the module docs for `vmware_host`, which is what this role uses. It does seem like theres a lot of redundant options but based on the docs:
A user might use the `reconnect` state if they had a host that may not exist, and they dont want to try adding it. 
A user might use the `present` state when they want to add a new host, but dont want to reconnect it if it exists.

I think using `vcenter_host_connection_state=present   vcenter_host_connection_reconnect_disconnected=True` is the same as `vcenter_host_connection_state=add_or_reconnect` though. So i could remove the documentation for add_or_reconnect?